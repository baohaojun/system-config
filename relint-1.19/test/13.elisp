;;; Relint test file 13          -*- emacs-lisp -*-

;; Test ineffective backslashes in strings

(defun f1 ()
  "doc string:
\! \( \) \[ \] \' all ignored
not ignored: \{ \} \| \` \? \* \. \+ \; \q etc"
  (list
   "\more"
   "not ignored: \( \) \[ \] \'"
   "long string \;
\; and so on"
   "valid: \\ \" \x5e \172 \u1234 \u00000041 \C-x \M-e \ \
and \a \b \f \n \r \t \v \d \e \s \^P"))

(defun f2 ()
  (re-search-forward "\$\.x\+\+")
  '("bracketed \q string"))
