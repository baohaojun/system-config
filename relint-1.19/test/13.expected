13.elisp:8:14: Ineffective string escape `\{'
13.elisp:8:17: Ineffective string escape `\}'
13.elisp:8:20: Ineffective string escape `\|'
13.elisp:8:23: Ineffective string escape `\`'
13.elisp:8:26: Ineffective string escape `\?'
13.elisp:8:29: Ineffective string escape `\*'
13.elisp:8:32: Ineffective string escape `\.'
13.elisp:8:35: Ineffective string escape `\+'
13.elisp:8:38: Ineffective string escape `\;'
13.elisp:8:41: Ineffective string escape `\q'
13.elisp:10:5: Ineffective string escape `\m'
13.elisp:11:18: Ineffective string escape `\('
13.elisp:11:21: Ineffective string escape `\)'
13.elisp:11:24: Ineffective string escape `\['
13.elisp:11:27: Ineffective string escape `\]'
13.elisp:11:30: Ineffective string escape `\''
13.elisp:12:17: Ineffective string escape `\;'
13.elisp:18:23: In call to re-search-forward: Unescaped literal `$' (pos 0)
  "$.x++"
   ^
13.elisp:18:23: Ineffective string escape `\$'
13.elisp:18:25: Ineffective string escape `\.'
13.elisp:18:28: Ineffective string escape `\+'
13.elisp:18:30: In call to re-search-forward: Repetition of repetition (pos 4)
  "$.x++"
   ....^
13.elisp:18:30: Ineffective string escape `\+'
13.elisp:19:16: Ineffective string escape `\q'
